geoh5py/objects/object_base.py

```diff
@@ -386,11 +392,11 @@ def find_or_create_property_group(self, **kwargs) -> PropertyGroup:
         """
 
         prop_group = None
-        if "name" in kwargs:
-            prop_group = self.get_property_group(kwargs["name"])[0]
+        if name is not None or uid is not None:
```

Comment:
what if the user pass a name of an existing property group (or multiple property groups exist with the same name) and another uid? We should  check if uid, then for name.

Reply:
We are not allowing two property groups to have the same name - it makes GA angry. But I like the idea of prioritizing uuid. 

Done